ext/standard/string.c

```diff
@@ -5453,8 +5453,8 @@ PHP_FUNCTION(substr_count)
 			length += (haystack_len - offset);
 		}
 		if (length < 0 || ((size_t)length > (haystack_len - offset))) {
-			php_error_docref(NULL, E_WARNING, "Invalid length value");
-			RETURN_FALSE;
+			zend_argument_value_error(4, "must be contained in argument #1 ($haystack)");
```

Comment:
At one point with @nikic there was a discussion to not change the substr_* type of functions to ValueError don't know if this still holds